python/artm/artm_model.py

```diff
@@ -929,7 +929,9 @@ def transform(self, batch_vectorizer=None, theta_matrix_type='dense_theta',
         elif theta_matrix_type == 'cache':
             theta_matrix_type_real = const.ThetaMatrixType_Cache
 
-        batches_list = [batch.filename for batch in batch_vectorizer.batches_list]
+        batches_list = batch_vectorizer.batches_list
+        if not batch_vectorizer.process_in_memory:
+            batches_list = [batch.filename for batch in batch_vectorizer.batches_list]
 
```

Comment:
Looks good! 👍
Anyway, I have an idea about adding property in `BatchVectorizer` class which returns filenames or ids according to type of processing. Can it be better to hide `process_in_memory` state from external environment?